test/unit-tests/expression/function/evaluate.test.js

```diff
@@ -84,4 +84,101 @@ describe('evaluate', function () {
     assert.strictEqual(expr1.toTex(), '\\mathrm{evaluate}\\left( expr\\right)')
     assert.strictEqual(expr2.toTex(), '\\mathrm{evaluate}\\left( expr, scope\\right)')
   })
+
+  describe('nullish coalescing operator', function () {
+    it('should handle basic nullish coalescing', function () {
+      assert.strictEqual(math.evaluate('null ?? 42'), 42)
+      assert.strictEqual(math.evaluate('undefined ?? 42'), 42)
+      assert.strictEqual(math.evaluate('0 ?? 42'), 0)
+      assert.strictEqual(math.evaluate('false ?? 42'), false)
+      assert.strictEqual(math.evaluate('"" ?? 42'), '')
+      assert(isNaN(math.evaluate('NaN ?? 42'))) // Should return NaN, not 42
+    })
+
+    it('should handle nullish coalescing with variables', function () {
+      const scope1 = {}
+      assert.throws(() => math.evaluate('x ?? 42', scope1), /Undefined symbol x/)
+
+      const scope2 = { x: null }
+      assert.strictEqual(math.evaluate('x ?? 42', scope2), 42)
+
+      const scope3 = { x: 0 }
+      assert.strictEqual(math.evaluate('x ?? 42', scope3), 0)
+
+      const scope4 = { x: undefined }
+      assert.strictEqual(math.evaluate('x ?? 42', scope4), 42)
+
+      const scope5 = { x: 5 }
+      assert.strictEqual(math.evaluate('x ?? 42', scope5), 5)
+    })
+
+    it('should handle chained nullish coalescing', function () {
+      assert.strictEqual(math.evaluate('null ?? undefined ?? 42'), 42)
+      assert.strictEqual(math.evaluate('null ?? 10 ?? 42'), 10)
+      assert.strictEqual(math.evaluate('5 ?? null ?? 42'), 5)
+      assert.strictEqual(math.evaluate('null ?? null ?? null ?? 99'), 99)
+    })
+
+    it('should handle nullish coalescing with correct precedence', function () {
+      // ?? has lower precedence than arithmetic operators
+      assert.strictEqual(math.evaluate('null ?? 1 + 2'), 3) // null ?? (1 + 2)
+      assert.strictEqual(math.evaluate('null ?? 2 * 3'), 6) // null ?? (2 * 3)
+      assert.strictEqual(math.evaluate('null ?? 2^3'), 8) // null ?? (2^3)
```

Comment:
These tests do not actually test the precedence, because `(null ?? 1) + 2` also evaluates to 3, and so on for the others. There's not any problem with having these tests, but please add tests where the behavior actually depends on the precedence. Thanks.